### DISCUSSION TITLE
How to model a series elastic actuator?

### DISCUSSION THREAD

**Discussion by v-r-a** [google-deepmind/mujoco#226] [Asking for Help] [answered]
I wish to model a torsional series elastic actuator. The connection may be defined as follows:
Parent link--motor--gearbox--torsion spring--child link.

Do we use the same trick as [illustrated for backlash](https://mujoco.readthedocs.io/en/latest/modeling.html#backlash) (with joint limits removed/expanded to real limits and added joint stiffness)?

I wish to know the most efficient way since I am working on a humanoid.

**Comment by yuvaltassa** [comment, marked as answer]
The most efficient thing would be to have only one joint and to add armature to this joint in order to simulate the inertia reflected through the gearbox. The other option would indeed be to have two serially connected parallel hinges. This option would be more expressive, but possibly too expressive, as the middle link (corresponding to the motor) could move independently of its parent and child. Basically the first link would need to have high PD gains for this to be realistic. I would probably try both and maybe do some system identification.
